### PR TITLE
[Merged by Bors] - feat: quasi-isomorphisms are preserved by functors which preserve homology

### PR DESCRIPTION
**Pull request by joelriou**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Homology/ShortComplex/PreservesHomology.lean [hunk 1]
```diff
@@ -720,6 +720,127 @@ noncomputable def homologyFunctorIso
...
+
 end ShortComplex
 
+namespace Functor
+
+variable (F : C ⥤ D) [F.PreservesZeroMorphisms] (S : ShortComplex C)
+
+/-- If a short complex `S` is such that `S.f = 0` and that the kernel of `S.g` is preserved
+by a functor `F`, then `F` preserves the left homology of `S`. -/
+noncomputable def preservesLeftHomologyOfOfZerof (hf : S.f = 0)
```
Comment (jcommelin) [inline Mathlib/Algebra/Homology/ShortComplex/PreservesHomology.lean:791]: I guess the `OfOf` is unintended
```suggestion
noncomputable def preservesLeftHomologyOfZerof (hf : S.f = 0)
```

File: Mathlib/Algebra/Homology/ShortComplex/PreservesHomology.lean [hunk 2]
```diff
@@ -720,6 +720,127 @@ noncomputable def homologyFunctorIso
...
+    [PreservesLimit (parallelPair S.g 0) F] :
+    F.PreservesLeftHomologyOf S := ⟨fun h =>
+  { g := by infer_instance
+    f' := Limits.preservesCokernelZero' _ _
+      (by rw [← cancel_mono h.i, h.f'_i, zero_comp, hf]) }⟩
+
+/-- If a short complex `S` is such that `S.g = 0` and that the cokernel of `S.f` is preserved
+by a functor `F`, then `F` preserves the right homology of `S`. -/
+noncomputable def preservesRightHomologyOfOfZerog (hg : S.g = 0)
```
Comment (jcommelin) [inline Mathlib/Algebra/Homology/ShortComplex/PreservesHomology.lean:800]:
```suggestion
noncomputable def preservesRightHomologyOfZerog (hg : S.g = 0)
```

File: Mathlib/Algebra/Homology/ShortComplex/PreservesHomology.lean [hunk 3]
```diff
@@ -720,6 +720,127 @@ noncomputable def homologyFunctorIso
...
+    [PreservesColimit (parallelPair S.f 0) F] :
+    F.PreservesRightHomologyOf S := ⟨fun h =>
+  { f := by infer_instance
+    g' := Limits.preservesKernelZero' _ _
+      (by rw [← cancel_epi h.p, h.p_g', comp_zero, hg]) }⟩
+
+/-- If a short complex `S` is such that `S.g = 0` and that the cokernel of `S.f` is preserved
+by a functor `F`, then `F` preserves the left homology of `S`. -/
+noncomputable def preservesLeftHomologyOfOfZerog (hg : S.g = 0)
```
Comment (jcommelin) [inline Mathlib/Algebra/Homology/ShortComplex/PreservesHomology.lean:809]:
```suggestion
noncomputable def preservesLeftHomologyOfZerog (hg : S.g = 0)
```

File: Mathlib/Algebra/Homology/ShortComplex/PreservesHomology.lean [hunk 4]
```diff
@@ -720,6 +720,127 @@ noncomputable def homologyFunctorIso
...
+      infer_instance
+    f' := by
+      have := h.isIso_i hg
+      let e : parallelPair h.f' 0 ≅ parallelPair S.f 0 :=
+        parallelPair.ext (Iso.refl _) (asIso h.i) (by aesop_cat) (by aesop_cat)
+      exact Limits.preservesColimitOfIsoDiagram F e.symm}⟩
+
+/-- If a short complex `S` is such that `S.f = 0` and that the kernel of `S.g` is preserved
+by a functor `F`, then `F` preserves the right homology of `S`. -/
+noncomputable def preservesRightHomologyOfOfZerof (hf : S.f = 0)
```
Comment (jcommelin) [inline Mathlib/Algebra/Homology/ShortComplex/PreservesHomology.lean:823]:
```suggestion
noncomputable def preservesRightHomologyOfZerof (hf : S.f = 0)
```

**Review by jcommelin — APPROVED** [review]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

**Comment by joelriou** [comment]
bors merge

**Pull request closed without merging**